### ISSUE TITLE
Hubot's brain is reset when I change rooms

### ISSUE THREAD

**Issue by kevinkuszyk** [RocketChat/hubot-rocketchat#178]
I'm using [hubot-azure-scripts](https://github.com/bfcamara/hubot-azure-scripts) to persist my Hubot's brain into Azure blob storage, and I have some code like this to save custom attributes to users:

```coffeescript
msg.message.user.teamcity.username = teamcityLogin
```

Everything is working well, and I can see my custom attributes saved into the `brain-dump.json` file:

```json
{  
   "users":{  
      "PP4JXZLw5pdWdWdDs":{  
         "id":"PP4JXZLw5pdWdWdDs",
         "name":"kevin",
         "room":"qwKmzsBrwouvAY3Br",
         "teamcity":{  
            "username":"kevin"
         }
      }
   },
}
```

However when I change rooms and send the bot a message, my custom attributes are lost:

```json
{
   "users":{
      "PP4JXZLw5pdWdWdDs":{
         "id":"PP4JXZLw5pdWdWdDs",
         "name":"kevin",
         "room":"Zy2ew95zTKhxyd3uj"
      }
   }
}
```

Have you seen anything like this before, and can you offer a steer on what I might do to fix it?  Thanks!

**Comment by geekgonecrazy** [comment]
How are you actually saving or adding to the hubot brain?

**Comment by kevinkuszyk** [comment]
Here's my full method to save a user's team city username (it's based on [this method](https://github.com/github/hubot-scripts/blob/master/src/scripts/github-credentials.coffee#L31-L34) in [github-credentials.coffee](https://github.com/github/hubot-scripts/blob/master/src/scripts/github-credentials.coffee):

```coffeescript
robot.respond /i am ([a-z0-9-]+) on team city/i, (msg) ->
  teamcityLogin = msg.match[1]

  if not msg.message.user.teamcity
     msg.message.user.teamcity = { }

  msg.message.user.teamcity.username = teamcityLogin
  msg.send "Ok, you are #{teamcityLogin} on Team City"
```

As you can see I don't explicitly call `robot.brain.save()` as my understanding is that Hubot does that itself.

**Comment by geekgonecrazy** [comment]
@kevinkuszyk what happens if you explicitly call save?   Any change in behaviour?   hubot brain isn't something I really use.  So i'm not super familiar.

**Comment by kevinkuszyk** [comment]
@geekgonecrazy explicitly calling `save()` didn't make any difference (my custom data was already saved in the brain when I changed rooms).

After some digging I tracked down a bug in the adapter when a user changes room.  I'll close this issue now, as there's a fix in #185.
